lib/exercism/user_tracks_summary.rb

```diff
@@ -0,0 +1,81 @@
+require_relative '../../x'
+
+class UserTracksSummary
+  def self.call(user)
+    new(user).tracks_summary
+  end
+
+  attr_reader :user
+
+  def initialize(user)
+    @user = user
+  end
+
+  def tracks_summary
+    @tracks_summary ||= languages_contributed.map do |lan|
+      TrackSummary.new(lan, total_contribution_by_language(lan))
+    end
+  end
+
+  private
+
+  def languages_contributed
+    @languages_contributed ||= completed_exercises_hash.keys.concat(reviewed_exercises_hash.keys).uniq
```

Comment:
Do we really need to memoize this? Is this being called somewhere besides `def tracks_summary`?